apps/extension/src/shared/components/default-frontend-form/new-frontend-input.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  FocusEventHandler,
+  ForwardedRef,
+  forwardRef,
+  KeyboardEventHandler,
+  MutableRefObject,
+  useRef,
+  useState,
+} from 'react';
+import { SelectList } from '@penumbra-zone/ui/components/ui/select-list';
+import { cn } from '@penumbra-zone/ui/lib/utils';
+import { isValidUrl } from '../../utils/is-valid-url';
+
+const getUrlValidity = (url: string, initialUrl?: string): boolean => {
+  return isValidUrl(url) && url !== initialUrl;
+};
+
+const mergeRefs = <EL extends HTMLElement>(
+  ...refs: (MutableRefObject<EL> | ForwardedRef<EL>)[]
+) => {
+  return (node: EL) => {
+    for (const ref of refs) {
+      if (ref && typeof ref === 'object') {
+        ref.current = node;
+      }
+    }
+  };
+};
+
+interface NewFrontendInputProps {
+  defaultFrontend?: string;
+  selected: boolean;
+  onSelect: (url: string) => void;
+}
+
+export const NewFrontendInput = forwardRef<HTMLInputElement, NewFrontendInputProps>(
```

Comment:
The RPC selector step has a need for a custom field as well. This feels like it would be a good candidate to extract the custom input out as a shared component. What do you think?

Reply:
I made the custom RPC input look the same as this one. Still, they have different logic for now, which we can possibly unify in the future